examples/prediction/xgboost-train-and-deploy-in-gcp.ipynb

```diff
@@ -0,0 +1,300 @@
+    "\n",
+    "    def train(self):\n",
+    "        (train_X, train_y), (test_X, test_y) = read_input(self.train_input)\n",
+    "        model = train_model(train_X,\n",
+    "                          train_y,\n",
+    "                          test_X,\n",
+    "                          test_y,\n",
+    "                          self.n_estimators,\n",
+    "                          self.learning_rate)\n",
+    "\n",
+    "        eval_model(model, test_X, test_y)\n",
+    "        save_model(model, self.model_file)\n",
+    "        gcs_copy(self.model_file, self.gcs_model_file)\n",
+    "        print(\"Model saved to {}\".format(self.gcs_model_file))\n",
+    "\n",
+    "    def predict(self, X, feature_names):\n",
+    "        \"\"\"Predict using the model for given ndarray.\"\"\"\n",
+    "        #return [[1.1, 1.2]]\n",
+    "        if not self.model:\n",
+    "            gcs_copy(self.gcs_model_file, self.model_file)\n",
+    "            self.model = joblib.load(self.model_file)\n",
+    "        prediction = self.model.predict(data=X)\n",
+    "        return [[prediction.item(0), prediction.item(0)]]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training Locally"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HousingServe().train()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fairing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import fairing\n",
+    "\n",
+    "# Setting up google container repositories (GCR) for storing output containers\n",
+    "# You can use any docker container registry istead of GCR\n",
+    "GCP_PROJECT = fairing.cloud.gcp.guess_project_name()\n",
+    "DOCKER_REGISTRY = 'gcr.io/{}/fairing-job'.format(GCP_PROJECT)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build base image for remote training and prediction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "py_version = \".\".join([str(x) for x in sys.version_info[0:3]])\n",
+    "base_image = 'gcr.io/{}/fairing-predict-example:latest'.format(GCP_PROJECT)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!docker build --build-arg PY_VERSION={py_version} . -t {base_image}\n",
+    "!docker push {base_image}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training in KF"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "fairing.config.set_builder('docker', registry=DOCKER_REGISTRY, base_image=base_image)\n",
```

Comment:
We don't really want the docker builder in the notebook - this won't work in Kubeflow.

Use the append builder or cluster builder.

Reply:
This example is meant to be executed locally. I renamed the file to show that this is for notebooks running locally. I will add another example to work in KF notebooks. Probably using the high level APIs.